services/bettersocial/chat/allAnonChat.js

```diff
@@ -0,0 +1,129 @@
+const {User, ChatAnonUserInfo} = require('../../../databases/models');
+const {Op} = require('sequelize');
+const crypto = require('crypto');
+const BetterSocialConstantListUtils = require('../constantList/utils');
+
+const generate_channel_id_for_anon_chat = (owner, member) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(owner, member);
+  return hash.digest('hex');
+};
+
+const is_all_anon_user = async (members) => {
+  // check all members users are anonymous
+  let users = await User.findAll({
+    where: {
+      user_id: {
+        [Op.in]: members
+      },
+      is_anonymous: false
+    }
+  });
+  return users.length === 0 ? true : false;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `is_all_anon_user` function, you are checking if all users are anonymous by querying for non-anonymous users and then checking if the result length is zero. This approach can be inefficient if there are many non-anonymous users in the database. Instead, you could query for any non-anonymous user in the members list and return false as soon as one is found.

```diff
- let users = await User.findAll({
+ let user = await User.findOne({
    where: {
      user_id: {
        [Op.in]: members
      },
      is_anonymous: false
    }
  });
- return users.length === 0 ? true : false;
+ return user ? false : true;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->